_posts/2020-09-11-liver-patient-classification/2020-09-11-liver-patient-classification.Rmd

```diff
@@ -0,0 +1,717 @@
+# Metric features by class after log-transforming right-skewed features
+liver_df_log = liver_df %>%
+  mutate( # Log for features with skewed distributions
+    alkaline_phosphatase = log(alkaline_phosphatase), 
+    aspartate_transaminase = log(aspartate_transaminase),
+    direct_bilirubin = log(direct_bilirubin),
+    total_bilirubin = log(total_bilirubin),
+    alanine_transaminase = log(alanine_transaminase)
+  )
+  
+plot_boxplot(liver_df_log, by = "class", nrow = 3, ncol = 3, 
+             ggtheme = theme_minimal(base_size = 14))
+```
+
+```{r density, include=FALSE}
+# Density plots by class for metric features
+ggplot(data = gather(liver_df[, -2], "variable", "value", 1:9), 
+       aes(x = value, fill = class, colour = class)) +
+  stat_density() +
+  labs(x = "Value", y = "Density") +
+  facet_wrap(~variable, scales = "free") + 
+  theme_minimal()
+```
+
+Note, that the same result can be achieved more easily by using `r ref("PipeOpMutate")` from mlr3pipelines.
+This PipeOp provides a smooth implementation to scale numeric features for mlr3 tasks.
+
+## Correlation
+
+As we mentioned in the description of the data, there are features that are indirectly measured by another one.
+This suggests that they are highly correlated.
+Some of the models we want to compare assume independent features or have problems with multicollinearity.
+Therefore, we checked for correlations between features.
+
+```{r correlation, echo=FALSE, out.width='88%', fig.align='center'}
+# Correlation heatmap
+# Remove missing values and change factor variables to numeric ones (no impact on calculated correlations)
+liver_df_numeric = liver_df %>%
+  na.omit() %>%
+  mutate(
+    gender = as.numeric(ifelse(gender == "female", 1, 0)),
+    class = as.numeric(ifelse(class == "disease", 1, 0))
+  )
+
+corr = cor(liver_df_numeric)
+
+ggcorrplot(corr, type = "lower", legend.title = "Correlation", show.diag = TRUE, 
+           lab = TRUE, tl.cex = 7, lab_size = 2.8)
+```
+
+For four of the pairs we obtained a very high correlation coefficient. 
+Looking at these features, it is clear they affect each other.
+As the complexity of the model should be minimized and due to multicollinearity concerns, we decided to take only one of each pair.
+When deciding on which features to keep, we chose those that are more specific and relevant regarding liver disease.
+Therefore, we chose albumin over the ratio between albumin and globulin and also over the total amount of protein.
+The same argument applies to using the amount of direct bilirubin instead of the total amount of bilirubin.
+Regarding aspartate transaminase and alanine transaminase, it was not clear which one to use, especially since we have no given real world implementation for the task and no medical training.
+Since we did not notice any fundamental differences in the data for these two features, we arbitrarily chose aspartate transaminase.
+
+## Final Dataset
+
+```{r reduceAndScale, include=FALSE}
```

Comment:
The first `mutate` is ok as-is: Our transformations are not data-dependent.
The second `mutate_at` is actually problematic, as we do not differentiate between train and test data there.
So the second mutate needs to be re-written as a `PipeOp`. Please also add a sentence that explains this, and that this is important!